plugins/thunderbolt/fu-self-test.c

```diff
@@ -338,6 +348,36 @@ mock_tree_node_have_fu_device (const MockTree *node, gpointer data)
 	return node->fu_device != NULL;
 }
 
+static void
+write_controller_fw (const gchar *nvm)
+{
+	g_autoptr(GFile) nvm_device = NULL;
+	g_autoptr(GFile) nvmem = NULL;
+	g_autoptr(GOutputStream) os = NULL;
+	gssize n;
+	const guint8 fw[] = {
+		#include "test.controller.fw.hex"
```

Comment:
I used it locally after dumping the nvm_active info from the controller but can't upload it of course.
This will be changed to just copy the FW image file we'll use skipping the headers.

Reply:
@superm1 Can I use one of the images you uploaded here? https://github.com/dell/thunderbolt-nvm-linux/tree/master/payloads
I'm asking as @hughsie mentioned something about the file licensing, and maybe those are already cleared to be used here.

Reply:
Including like this is not going to work super well. I think just including the binary file itself in `data/tests` is much better, and then it can be loaded using `fu_test_get_filename()` and then `g_file_get_contents()`

Reply:
The setup function of the test must also be adjust to use a valid image, otherwise the upgrade test, which does the image verification implicitly, will fail. See [here](https://github.com/hughsie/fwupd/pull/201/files#diff-c3bcb97a5c09bf8f8e1e231225273e8eL851)

I think adding another flag for the test `TEST_INVALID_FIRMWARE` could be added that loads invalid firmware and then we can add a test verifying the verification works ;)

Reply:
I'm not so sure about distributing the Dell ones by #include will be kosher.  In data/tests might be OK, but not sure what license that would be under and if it would be compatible to the project.

Another idea is to just run tests in CI by wget in gnome-desktop-testing by wget'ing from that repo (or setting up another one to store stuff like this) so that you don't have to get into distribution issues.

Reply:
The wget isn't going to work with Fedora or RHEL as the builders (that run the tests) have no internet access.

Reply:
@superm1 But I didn't meant to `#include` the FW. This was just a quick-and-dirty solution for testing it locally. I just want to add it to `data/tests` as @hughsie mentioned above. Is this still a licensing issue?